client/app/components/queries/visualization-embed.js

```diff
@@ -34,7 +34,11 @@ const VisualizationEmbed = {
 
     document.querySelector('body').classList.add('headless');
 
-    this.refreshQueryResults();
+    if (this.query.is_safe) {
+      this.refreshQueryResults();
+    } else {
+      this.error = "Can't embed queries with text parameters.";
```

Comment:
I'd avoid being too specific on what are unsafe queries here, as the definition is subject to change soon and this message will likely be left unattended.

Reply:
I doubt it will change so soon... and until we have documentation about what are safe parameters, it's better to have a meaningful message.